### DISCUSSION TITLE
difference between imtime_bubble_chi0_wk and  lindhard_chi00_wk

### DISCUSSION THREAD

**Discussion by Lanlin123** [TRIQS/triqs#862] [Q&A]
Dear Triqs Developers,
     I find the bare bubble chi0's caculated by imtime_bubble_chi0_wk and  lindhard_chi00_wk are different. The  imtime_bubble_chi0_wk will give me negative chi0's in some μ levels while lindhard_chi00_wk not. But the chi0's calculated by lindhard_chi00_wk don't vary with the temperature beta. I am doing a  linearized Eliashberg equation calculation. I wonder which function you recommend to use. Are these negative chi0's calculated by imtime_bubble_chi0_wk physical?

Best regards,
Lanlin

**Comment by the-hampel** [comment]
Hi, 
sorry I am not an expert in this but I think one should be careful here. I think the `imtime_bubble_chi0_wk` is a numerical calculation and especially at lower temperature is hard to converge with respect to freq mesh and k points. The Lindhard formula is analytic and hence should be fine to use but is slower (https://triqs.github.io/tprf/latest/reference/python_reference.html#lindhard-non-interacting-generalized-susceptibility) . 

Did you read the tutorial: https://triqs.github.io/tprf/latest/user_guide/Square%20lattice%20susceptibility.html#Bare-generalized-susceptibility-\chi_0 which compares both equations. On the bottom of the tutorial it becomes clear that both do not give the exact same value. Maybe @HugoStrand or @YannIntVeld  can comment more on this, but I think that given that you know the Lehmann representation of your Green function, the Lindhard equation should give liable results, however I am a bit puzzled by the fact that you do not see any temperature effect. 

Best, 
Alex
